### PR TITLE
seqcli sample setup

### PR DESCRIPTION
**Pull request by nblumhardt**
Work-in-progress on a sample data loader, #133.

The plan is for two CLI commands:

 * `seqcli sample setup` - load static configuration like signals, queries, dashboards, etc.
 * `seqcli sample ingest` - send simulated event data in real-time

`setup` can be run once, tweaks made, and the whole thing brought to life by running and re-running the `ingest` command as required.

The PR's currently focused on the setup step. Rather than create entities programmatically, as we've done before in various internal tools, the sample is going to be based on entities exported from the Seq API in JSON format. The aim is to make the whole thing easier to update with quality content as Seq features are added, removed, and changed.

The entities in a Seq installation depend on each other in various ways; when a signal is created, it is assigned an id, and any dashboard or retention policy using that signal will have to reference the assigned id.

The sample loader achieves this using a simple templating system for the JSON input files. The JSON inputs are mostly regular JSON, but with the addition of function-call-like template value replacement; there's only one function supported, `ref()`, that resolves to the id assigned to an entity in an accompanying template file:

```
    "SignalId": ref("signal-HTTP Requests.json"),
```

In the long run, we're expecting to expose the whole template-import feature as a way to share content between Seq installations, and for us to publish useful templates like "ASP.NET Core request metrics" or "health check app with alerts and dashboards". At that point we expect to extend the available "functions" to include things like resolving an entity by name, or to parameters passed into something like a `seqcli template import` command.

### CODE REVIEW

**Comment by nblumhardt** [comment]
As things stand, `seqcli sample setup` produces a dashboard "ASP.NET", and two signals that the dashboard uses (shown below):

image

Once the sample event data is sorted out these will probably be replaced/made relevant to the demo.

**Comment by nblumhardt** [comment]
The `seqcli sample setup` command creates a bunch of entities, grouped under a _Sample_ workspace. The `seqcli sample ingest` command produces the data that lines up with the entities:

Sample event data:

image

Retention policy for sample data:

image

Database dashboard:

image

Web dashboard:

image

Ordering pipeline dashboard:

image

Going to clean up and call this "v1", out of time to develop it much further this week.

File: src/Roastery/Web/NetworkLatencyMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Roastery.Web
+{
+    class NetworkLatencyMiddleware : HttpServer
+    {
+        readonly HttpServer _next;
+        readonly Random _rng = new Random(DateTime.Now.Millisecond);
```
Comment (KodrAus) [inline src/Roastery/Web/NetworkLatencyMiddleware.cs:9]: This isn't thêaäfgr

segfault

**Review by KodrAus — APPROVED** [review]
Looks good to me!

**Pull request merged**